File: toeic-study-zone-be/src/main/java/com/toeicstudyzone/service/CommentService.java
package com.toeicstudyzone.service;

import com.toeicstudyzone.dto.request.CommentDTO;
import com.toeicstudyzone.entity.Comment;
import java.util.List;
import java.util.Optional;

public interface CommentService {
    Comment createComment(Comment comment);
    List<CommentDTO> getCommentsByTestId(Long testId); 
    List<CommentDTO> getRepliesByParentId(Long parentId); 
    Optional<Comment> updateComment(Long id, String commentText);
    void deleteComment(Long id);
    Optional<Comment> findById(Long id);
}

File: toeic-study-zone-fe/src/services/commentService.js
import api from "./api";

const commentService = {
  postComment(commentData) {
    // commentData = { userId, testId, commentText, parentId }
    return api.post("/api/comments", commentData);
  },

  getCommentsByTestId(testId) {
    return api.get(`/api/comments/test/${testId}`);
  },
};

export default commentService;
